astroquery/mast/core.py

```diff
@@ -145,24 +145,90 @@ def __init__(self, username=None, password=None, session_token=None):
 
         self._column_configs = dict()
         self._current_service = None
 
+        self._auth_mode = self._get_auth_mode()
+
+
+        if "SHIB-ECP" == self._auth_mode:
+            log.debug("Using Legacy Shibboleth login")
+            self._SESSION_INFO_URL = conf.server + "/Shibboleth.sso/Session"
+            self._SP_TARGET = conf.server + "/api/v0/Mashup/Login/login.html"
+            self._IDP_ENDPOINT = conf.ssoserver + "/idp/profile/SAML2/SOAP/ECP"
+            self._MAST_DOWNLOAD_URL = conf.server + "/api/v0/Download/file"
+        elif "MAST-AUTH" == self._auth_mode:
+            log.debug("Using Auth.MAST login")
+            self._SESSION_INFO_URL = conf.server + "/whoami"
+            self._MAST_DOWNLOAD_URL = conf.server + "/api/v0.1/Download/file"
+            self._MAST_BUNDLE_URL = conf.server + "/api/v0.1/Download/bundle"
+        else:
+            raise Exception("Unknown MAST Auth mode %s" % self._auth_mode)
+
         if username or session_token:
             self.login(username, password, session_token)
 
-    def _attach_cookie(self, session_token):  # pragma: no cover
+    def _get_auth_mode(self):
+        _auth_mode = "SHIB-ECP"
+
+        # Detect auth mode from auth_type endpoint
+        resp = self._session.get(conf.server + '/auth_type')
+        if resp.status_code == 200:
+            _auth_mode = resp.text.strip()
+        else:
+            log.warning("Unknown MAST auth mode, defaulting to Legacy Shibboleth login")
+        return _auth_mode
+
+    def _login(self, *args, **kwargs):
+        if "SHIB-ECP" == self._auth_mode:
+            return self._shib_legacy_login(*args, **kwargs)
+        elif "MAST-AUTH" == self._auth_mode:
+            return self._authorize(*args, **kwargs)
+        else:
+            raise Exception("Unknown MAST Auth mode %s" % self._auth_mode)
+
+    def get_token(self, *args, **kwargs):
+        """
+        Returns MAST token cookie.
+
+        Returns
+        -------
+        response : `~http.cookiejar.Cookie`
+        """
+        if "SHIB-ECP" == self._auth_mode:
+            return self._shib_get_token(*args, **kwargs)
+        elif "MAST-AUTH" == self._auth_mode:
+            return self._get_token(*args, **kwargs)
+        else:
+            raise Exception("Unknown MAST Auth mode %s" % self._auth_mode)
+
+    def session_info(self, *args, **kwargs):
+        """
+        Displays information about current MAST user, and returns user info dictionary.
+
+        Parameters
+        ----------
+        silent : bool, optional
+            Default False.
+            Suppresses output to stdout.
+
+        Returns
+        -------
+        response : dict
+        """
+        if "SHIB-ECP" == self._auth_mode:
+            return self._shib_session_info(*args, **kwargs)
+        elif "MAST-AUTH" == self._auth_mode:
+            return self._session_info(*args, **kwargs)
+        else:
+            raise Exception("Unknown MAST Auth mode %s" % self._auth_mode)
```

Comment:
I think it would be useful to raise more specific exceptions rather than the super generic `Exception`. This is not urgent though, and can be done in a follow-up PR.